src/Aserto.AspNetCore.Middleware/Clients/Directory/V3/IDirectoryAPIClient.cs

```diff
@@ -0,0 +1,161 @@
+
+        // Writer methods
+
+        /// <summary>
+        /// Updates an object if it exists, creates one if it doesn't exist.
+        /// </summary>
+        /// <param name="key">The key of the object.</param>
+        /// <param name="type">The type of the object.</param>
+        /// <param name="displayName">The display name of the object.</param>
+        /// <param name="properties">A struct representing the properties bag of the object.</param>
+        /// <param name="hash">The hash of the object.</param>
+        /// <returns>A SetObjectResponse object.</returns>
+        Task<SetObjectResponse> SetObjectAsync(string key, string type, string displayName = "", Struct properties = null, string hash = "");
+
+        /// <summary>
+        /// Deletes an object.
+        /// </summary>
+        /// <param name="key">The key of the object.</param>
+        /// <param name="type">The type of the object.</param>
+        /// <param name="withRelations">Remove the relations for this object also.</param>
+        /// <returns>A DeleteObjectResponse object.</returns>
+        Task<DeleteObjectResponse> DeleteObjectAsync(string key, string type, bool withRelations = false);
+
+        /// <summary>
+        /// Creates a relation if it doesn't exists, updates an existing one.
+        /// </summary>
+        /// <param name="subjectKey">The key of the subject.</param>
+        /// <param name="subjectType">The type of the subject.</param>
+        /// <param name="subjectRelation">The relaton of the subject.</param>
+        /// <param name="objKey">The key of the object.</param>
+        /// <param name="objType">The type of the object.</param>
+        /// <param name="relationName">The type name of the relation.</param>
+        /// <param name="hash">The hash of the object.</param>
+        /// <returns>A SetRelationResponse object.</returns>
+        Task<SetRelationResponse> SetRelationAsync(string subjectKey, string subjectType, string subjectRelation, string objKey, string objType, string relationName, string hash = "");
+
+        /// <summary>
+        /// Deletes a relation.
+        /// </summary>
+        /// <param name="subjectKey">The key of the subject.</param>
+        /// <param name="subjectType">The type of the subject.</param>
+        /// <param name="subjectRelation">The relaton of the subject.</param>
+        /// <param name="objKey">The key of the object.</param>
+        /// <param name="objType">The type of the object.</param>
+        /// <param name="relationName">The name of the relation type.</param>
+        /// <param name="relationObjectType">The type of the object indicated by relation.</param>
+        /// <returns>A DeleteRelationResponse object.</returns>
+        Task<DeleteRelationResponse> DeleteRelationAsync(string subjectKey = "", string subjectType = "", string subjectRelation = "", string objKey = "", string objType = "", string relationName = "", string relationObjectType = "");
```

Comment:
I don't think the v3 directory has a `relationObjectType`

<img width="636" alt="Screenshot 2023-11-10 at 15 38 00" src="https://github.com/aserto-dev/aserto-dotnet/assets/3924120/ca1a0619-1e22-4b5a-ba83-a95bfd6797ba">